pkg/autodiscovery/configresolver.go

```diff
@@ -184,11 +186,13 @@ func (cr *ConfigResolver) resolve(tpl integration.Config, svc listeners.Service)
 	}
 
 	// store resolved configs in the AC
-	cr.ac.m.Lock()
-	defer cr.ac.m.Unlock()
-	cr.ac.loadedConfigs[resolvedConfig.Digest()] = resolvedConfig
+	cr.ac.store.setLoadedConfig(resolvedConfig)
 	cr.ac.store.addConfigForService(svc.GetID(), resolvedConfig)
 	cr.configToService[resolvedConfig.Digest()] = svc.GetID()
+	cr.ac.store.setTagsHashForService(
+		svc.GetID(),
+		tagger.GetEntityHash(docker.ContainerIDToEntityName(string(svc.GetID()))),
```

Comment:
Warn: services from the `kubelet` listener will already have a scheme, shall we only call `ContainerIDToEntityName` when there's none, like https://github.com/DataDog/datadog-agent/pull/1839/files#diff-6f38a99c24b92894c615272d9c304d1bR421 ?